python-backend/main.py

```diff
@@ -31,42 +33,74 @@
                    allow_headers=["*"])
 
 
+def parse(data: Input) -> Tuple[Mul, Add, Symbol]:
```

Comment:
The import of `Tuple` can be eliminated by changing the return value type to `tuple[Mul, Add, Symbol]` (unless you need to support older versions of python (I think that this was introduced in 3.7?)).